tidb: enhance clarity and ensure consistency for the system variable tidb_auto_analyze_concurrency (#20704)
pingcap/docs: system-variables.md

- Type: Integer
- Default value: `1`
- Range: `[1, 2147483647]`

> - This variable controls the number of concurrent auto-analyze operations that can run in a TiDB cluster. From v8.4.0, this concurrency was fixed at 1. To accelerate statistics collection tasks, you can increase this concurrency based on the available resources in your cluster.

Reviewer: ```suggestion
- This variable controls the number of concurrent auto-analyze operations that can run in a TiDB cluster. Before v8.4.0, this concurrency is fixed at 1. To accelerate statistics collection tasks, you can increase this concurrency based on the available resources in your cluster.
```
